api/objects.proto

```diff
@@ -257,6 +257,15 @@ message NetworkAttachment {
 	repeated string aliases = 3;
 }
 
+message CNMState {
+	// Driver specific operational state provided by the network driver.
+	Driver driver_state = 4;
+
+	// Runtime state of IPAM options. This may not reflect the
+	// ipam options from NetworkSpec.
+	IPAMOptions ipam = 5 [(gogoproto.customname) = "IPAM"];
```

Comment:
Right, I kept them the same as before while I tried to figure out the compat story. I'll fix that up as I do whatever it is I find I need to do once I've digested you other comments (particularly https://github.com/docker/swarmkit/pull/2108#issuecomment-293115890!)